src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssApi.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static class GssApi
+    {
+        internal static bool EstablishSecurityContext(
+            ref SafeGssContextHandle context,
```

Comment:
I think I'd rather see this either a) always allocate the handle (maybe rename to CreateSecurityContext) or b) always initialize an existing handle (rename to InitializeSecurityContext?).  Having this be dual-purpose like this seems like it will lead to confusion.